### DISCUSSION TITLE
Perf of nginx with different openssl versions (1.1.1 vs 3.0.9)

### DISCUSSION THREAD

**Discussion by velkejondra** [openssl/openssl#21652] [Q&A]
Hi.

I performed perf of nginx with openssl1.1.1 and openssl3.0.9. In addition to significantly more time spent in rwlock operations in openssl3.0.9, for which there are already issues. The 3.0.9 version also spends 10 percent of the whole perf in function x25519_fe64_sub, while this function doesn't show up at all in openssl1.1.1 perf.

Is this normal behavior?

Thanks for your answers.

**Comment by arapov** [comment]
Hi, @velkejondra!
We have a "[tracking issue](https://github.com/openssl/openssl/issues/20286)" for performance problems occurring between v1.1 and v3.0. Please check to see if your issue is described there. If it's not, we would greatly appreciate a report from you, ideally including details that would enable us to reproduce the issue.

We do pay close attention to performance issues and have resolved a number of them in v3.1 and the upcoming release.
